wallet/createtx_test.go

```diff
@@ -449,42 +449,105 @@ func TestSelectUtxosTxoToOutpoint(t *testing.T) {
 	require.Len(t, unspent, 4, "expected 4 unspent "+
 		"utxos")
 
-	selectUtxos := []wire.OutPoint{
+	tCases := []struct {
+		name        string
+		selectUtxos []wire.OutPoint
+		errString   string
+	}{
 		{
-			Hash:  incomingTx.TxHash(),
-			Index: 1,
+			name: "Duplicate utxo values",
+			selectUtxos: []wire.OutPoint{
+				{
+					Hash:  incomingTx.TxHash(),
+					Index: 1,
+				},
+				{
+					Hash:  incomingTx.TxHash(),
+					Index: 1,
+				},
+			},
+			errString: "selected UTXOs contain duplicate values",
 		},
 		{
-			Hash:  incomingTx.TxHash(),
-			Index: 2,
+			name: "all select utxo not part of unspent utxos",
+			selectUtxos: []wire.OutPoint{
+				{
+					Hash:  chainhash.Hash([32]byte{1}),
+					Index: 1,
+				},
+				{
+					Hash:  chainhash.Hash([32]byte{3}),
+					Index: 1,
+				},
+			},
+			errString: "selected outpoint not eligible for spending",
+		},
+		{
+			name: "part select utxo not part of unspent utxos",
+			selectUtxos: []wire.OutPoint{
+				{
+					Hash:  chainhash.Hash([32]byte{1}),
+					Index: 1,
+				},
+				{
+					Hash:  incomingTx.TxHash(),
+					Index: 1,
+				},
+			},
+			errString: "selected outpoint not eligible for spending",
+		},
+		{
+			name: "select utxo, no error",
+			selectUtxos: []wire.OutPoint{
+				{
+					Hash:  incomingTx.TxHash(),
+					Index: 1,
+				},
+				{
+					Hash:  incomingTx.TxHash(),
+					Index: 2,
+				},
+			},
 		},
 	}
 
-	// Test by sending 200_000.
-	targetTxOut := &wire.TxOut{
-		Value:    200_000,
-		PkScript: p2trScript,
-	}
-	tx1, err := w.txToOutputs(
-		[]*wire.TxOut{targetTxOut}, nil, nil, 0, 1, 1000,
-		CoinSelectionLargest, true, selectUtxos, alwaysAllowUtxo,
-	)
-	require.NoError(t, err)
+	for _, tc := range tCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Test by sending 200_000.
+			targetTxOut := &wire.TxOut{
+				Value:    200_000,
+				PkScript: p2trScript,
+			}
+			tx1, err := w.txToOutputs(
+				[]*wire.TxOut{targetTxOut}, nil, nil, 0, 1, 1000,
+				CoinSelectionLargest, true, tc.selectUtxos, alwaysAllowUtxo,
```

Comment:
Can you please set up your IDE in a way that you get a visual indication of where the 80 characters cutoff limit is (and setting the tabulator width correctly too)?

Reply:
Apologies about this, as I just copied and put it in a for block, I did not recheck for the effect on the line length.

Reply:
Please I would like to ask, could you tell by just looking at it or you pulled it into your IDE, or you are using a different platform for review?